### PR TITLE
1890 jsdoc

### PR DESCRIPTION
**Pull request by IagoLast**

### CODE REVIEW

File: src/api/v4/dataview/base.js [hunk 1]
```diff
@@ -328,3 +330,23 @@ Base.prototype.$getInternalModel = function () {
 };
 
 module.exports = Base;
+
+/**
+ * Event triggered when the column in a dataview changes.
+ *
+ * Contains a single argument with the name of the changed column.
+ * 
+ * @event carto.dataview.Base.columnChanged
```
Comment (alonsogarciapablo) [inline src/api/v4/dataview/base.js]: From what I understand, [@event](http://usejsdoc.org/tags-event.html) is something you should use to document the object associated to an event, and not the event type itself. I might be wrong, but [here is how I used it](https://github.com/CartoDB/cartodb.js/blob/public-api/src/api/v4/leaflet/layer-group.js#L73-L82).

**Comment by Jesus89** [comment]
I think we need to use the @fires tag in the functions that triggers the events.

File: src/api/v4/dataview/base.js (hunk 1 above)
Comment (IagoLast) [reply src/api/v4/dataview/base.js]: I based my doc on [this example](http://usejsdoc.org/tags-event.html):

> A typical event is represented by an object with a defined set of properties.

So `carto.dataview.Base.columnChanged` is an event which is only a string with the column name.

**Comment by IagoLast** [comment]
@Jesus89 maybe you are right, but I want to emphasise the fact that events can arrive asynchronously, so is the object the one throwing events.

A client can throw a `success` without explicitly  calling `client.reload()`

File: src/api/v4/dataview/base.js (hunk 1 above)
Comment (alonsogarciapablo) [reply src/api/v4/dataview/base.js]:
```
  /**
     * Snowball event.
     *
     * @event Hurl#snowball
     * @type {object}
     * @property {boolean} isPacked - Indicates whether the snowball is tightly packed.
     */
    this.emit('snowball', {
        isPacked: this._snowball.isPacked
    });
```

`@event` is NOT documenting "snowball" (name of the event, like `columnChanged`). It's documenting `{ isPacked: xxx }`. I don't think we want `carto.dataview.Base.columnChanged` as a menu item in the documentation.

Comment (alonsogarciapablo) [reply src/api/v4/dataview/base.js]: Actually, perhaps we might en up having a single `ChangeEvent` for all `xxxChanged` events, don't you think?

**Pull request merged**